client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -84,12 +87,14 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
 
     public RawClient(PravegaNodeUri uri, ConnectionFactory connectionFactory) {
         this.segmentId = null;
+        this.connectionFactory = connectionFactory;
```

Comment:
Check for null. I suggest adding `@NonNull` as an annotation (mind the casing; don't confuse with `@Nonnull`)

Reply:
Same below

Reply:
Got it, thanks!